addons/payment_razorpay/views/payment_provider_views.xml

```diff
@@ -11,17 +11,54 @@
                        invisible="code != 'razorpay'">
                     <field name="razorpay_key_id"
                            string="Key Id"
-                           required="code == 'razorpay' and state != 'disabled'"/>
+                           groups="base.group_no_one"/>
                     <field name="razorpay_key_secret"
                            string="Key Secret"
-                           required="code == 'razorpay' and state != 'disabled'"
-                           password="True"/>
-                    <field name="razorpay_webhook_secret"
-                           string="Webhook Secret"
-                           required="code == 'razorpay' and state != 'disabled'"
+                           groups="base.group_no_one"
                            password="True"/>
+                    <label for="razorpay_webhook_secret"/>
+                    <div class="o_row">
+                        <field name="razorpay_webhook_secret"
+                            string="Webhook Secret"
+                            groups="base.group_no_one"
+                            password="True"/>
+                        <button string="Create Webhook"
+                                name="action_razorpay_create_and_update_webhook"
+                                type="object"
+                                help="If you change the your base url, you will need to update the webhook. else payment state will not be updated in odoo."
+                                class="btn-primary"
+                                groups="base.group_no_one"
+                                invisible="not razorpay_access_token or razorpay_webhook_id"/>
+                        <button string="Update Webhook"
```

Comment:
Since both button call same method you could make one button called `Refresh Webhook` IMHO :see_no_evil: 
Check with @dit-odoo